htdocs/core/tpl/extrafields_list_search_sql.tpl.php

```diff
@@ -23,7 +23,11 @@
 
 		if ($crit != '' && in_array($typ, array('date', 'datetime', 'timestamp')))
 		{
-			$sql .= " AND ".$extrafieldsobjectprefix.$tmpkey." = '".$db->idate($crit)."'";
+			if ($typ == 'date'){
+				$sql .= " AND ".$extrafieldsobjectprefix.$tmpkey." = DATE('".$db->idate($crit)."')";
```

Comment:
DATE function on database level must not be used : It does not share the same timezone as the PHP timezone that is mostly used by the main code.
So you must convert the $crit timestamp into a value where hour, minute and seconds are 0.
For this, you can replace  $crit  with  dol_get_first_hour($crit)